src/Compilers/CSharp/Portable/Parser/Blender.Reader.cs

```diff
@@ -305,6 +305,7 @@ private static bool IsFabricatedToken(SyntaxKind kind)
                 {
                     case SyntaxKind.GreaterThanGreaterThanToken:
                     case SyntaxKind.GreaterThanGreaterThanEqualsToken:
+                    case SyntaxKind.ExclamationExclamationToken:
```

Comment:
my bad on forgetting about this @Youssef1313 @RikkiGibson .  The merging tokens bit totally should have triggered things for me!

Reply:
Consider adding a note to `MergeAdjacent` to mention this method should be updated when used.

Reply:
> Consider adding a note to MergeAdjacent to mention this method should be updated when used.

It looks `LanguageParser.ParseLambdaExpression()` also uses `MergeAdjacent()` for `SyntaxKind.EqualsGreaterThanToken`.

Reply:
good catch!

Reply:
Consider adding an assert to `MergeAdjacent()`: `Debug.Assert(IsFabricatedToken(kind));`

Reply:
Logged [https://github.com/dotnet/roslyn/issues/60002](https://github.com/dotnet/roslyn/issues/60002).